### PR TITLE
chore: move initialization logic to light client module method

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

closes: #5974 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

## Summary by CodeRabbit

- **Refactor**
	- Enhanced the Solo Machine light client initialization process by removing obsolete logic.
- **Bug Fixes**
	- Ensured accurate comparison of consensus states during client initialization for reliable setup.
- **Tests**
	- Updated and added tests to validate new initialization logic and consensus state handling in the Solo Machine light client.

### CODE REVIEW

File: modules/light-clients/06-solomachine/client_state.go [hunk 1]
```diff
@@ -67,19 +65,6 @@ func (cs ClientState) Validate() error {
 	return cs.ConsensusState.ValidateBasic()
 }
 
-// Initialize checks that the initial consensus state is equal to the latest consensus state of the initial client and
-// sets the client state in the provided client store.
-func (cs ClientState) Initialize(_ sdk.Context, cdc codec.BinaryCodec, clientStore storetypes.KVStore, consState exported.ConsensusState) error {
```
[the remark this answers is not in the document]
Comment (crodriguezvega) [reply modules/light-clients/06-solomachine/client_state.go:72]: I guess it doesn't hurt. Also, even though the migration docs already mention that the interface function has been removed, maybe it's worth adding a separate section in the migration docs to document that this function has also been removed from the concrete type.

File: modules/light-clients/06-solomachine/light_client_module_test.go [hunk 2]
```diff
@@ -13,6 +16,86 @@ const (
...
+				clienttypes.ErrInvalidConsensus,
+			},
+			{
+				"invalid client state: sequence is zero",
+				sm.ConsensusState(),
+				solomachine.NewClientState(0, sm.ConsensusState()),
+				clienttypes.ErrInvalidClient,
+			},
+			{
+				"invalid client state",
```
Comment (crodriguezvega) [inline modules/light-clients/06-solomachine/light_client_module_test.go:62]:
```suggestion
				"invalid client state: Tendermint client state",
```

**Review by crodriguezvega — APPROVED** [review]
Looks perfectly fine to me. I would still add a note in the migration docs.

**Review by chatton — APPROVED** [review]
Changes look good to me, happy to merge this with or without the changelog (can happen in a follow up)

**Review by damiannolan — APPROVED** [review]
LGTM

**Pull request merged**